modules/Timetable Admin/courseEnrolment_manage_byPerson_edit_editProcess.php

```diff
@@ -64,9 +65,11 @@
                     header("Location: {$URL}");
                 } else {
                     //Write to database
+                    $dateEnrolled = $role != $values['role'] && stripos($role, 'Left') === false ? date('Y-m-d') : $values['dateEnrolled'];
```

Comment:
Can you explain the relationship between role and un/enroll date? Not sure I get it!

Reply:
It's looking to see if the role has changed when submitting the form, eg: to prevent changing the enrol date when only the reportable flag has been changed. For this logic: if the role has changed and the new role is not left, then update the enrolled date, otherwise keep it the same. This happens when setting a left enrolment back to non-left.